### PR TITLE
Analytics screen

### PR DESCRIPTION
**Pull request by chaotixkilla**

### CODE REVIEW

**Comment by chaotixkilla** [comment]
[Analytics PT1](https://trello.com/c/3IyFGuPy/828-analytics-pt1)

File: new-lamassu-admin/src/pages/Analytics/Analytics.js [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+  mainCurrency
+}) => {
+  const classes = useStyles()
+
+  const isCurrency = !!currencies
+  // const mostUsedCurrency = R.reduce(
+  //   (acc, v) => (acc === '' ? v : currencies[v] > currencies[acc] ? v : acc),
+  //   '',
+  //   R.keys(currencies)
+  // )
```
Comment (josepfo) [inline new-lamassu-admin/src/pages/Analytics/Analytics.js]: Did you miss this code or intend to use it in the future?
```suggestion
```

**Review by josepfo — CHANGES_REQUESTED** [review]
Great job, looks awesome!
Just a few details, according to the spec the percentage on each OverviewEntry should be bold and without empty space between the icon. Also, the background on the Select drop-down should have a dark color.

**Review by josepfo — APPROVED** [review]

File: new-lamassu-admin/src/pages/Analytics/Analytics.js [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+import { fromNamespace } from 'src/utils/config'
+
+import styles from './Analytics.styles'
+import Graph from './Graph'
+
+const useStyles = makeStyles(styles)
+
+const DAY = 24 * 60 * 60 * 1000
+const WEEK = 7 * 24 * 60 * 60 * 1000
+const MONTH = 30 * 24 * 60 * 60 * 1000
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Analytics/Analytics.js]: Don't we use this elswhere?

File: new-lamassu-admin/src/pages/Analytics/Graph.js [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+  java,
+  neon,
+  subheaderDarkColor,
+  offColor,
+  fontColor,
+  primaryColor,
+  fontSecondary
+} from 'src/styling/variables'
+
+const MINUTE = 60 * 1000
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Analytics/Graph.js]: We do hehe, add this to a consts file on utils.

File: new-lamassu-admin/src/pages/Analytics/Graph.js [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+const DAY = 24 * 60 * 60 * 1000
+const WEEK = 7 * 24 * 60 * 60 * 1000
+const MONTH = 30 * 24 * 60 * 60 * 1000
+
+const Graph = ({ data, representing, period, timezone }) => {
+  const ref = useRef(null)
+
+  const GRAPH_HEIGHT = 401
+  const GRAPH_WIDTH = 1163
+  const GRAPH_MARGIN = useMemo(
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Analytics/Graph.js:22]: Is there a reason for this to be a memoized react component? Why not a simple object?

File: new-lamassu-admin/src/pages/Analytics/Graph.js [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+      'Apr',
+      'May',
+      'Jun',
+      'Jul',
+      'Aug',
+      'Sep',
+      'Oct',
+      'Nov',
+      'Dec'
+    ]
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Analytics/Graph.js]: I'm pretty sure we can get this from a lib, is the react project using any date libs?

File: new-lamassu-admin/src/pages/Analytics/Analytics.js [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+          d => new Date(d.created).getTime() >= Date.now() - MONTH
+        ) ?? [],
+      previous:
+        machineTxs.filter(
+          d =>
+            new Date(d.created).getTime() < Date.now() - MONTH &&
+            new Date(d.created).getTime() >= Date.now() - 2 * MONTH
+        ) ?? []
+    }
+  }
```
Comment (RafaelTaranto) [inline new-lamassu-admin/src/pages/Analytics/Analytics.js]: This would also get pretty simplified with a lib.

**Review by RafaelTaranto — CHANGES_REQUESTED** [review]
Also has conflicts

File: new-lamassu-admin/src/pages/Analytics/Graph.js (hunk 4 above)
Comment (chaotixkilla) [reply new-lamassu-admin/src/pages/Analytics/Graph.js:22]: Mostly for optimization purposes, as the `useEffect` hook is calling a function, which "needs" to be wrapped in `useCallback`, which calls these objects, which should be memoized ideally I think. Might not matter as much to require optimization

**Pull request closed without merging**